src/analysis/lattices/powerset2.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2023 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <unordered_set>
+
+#include "../lattice.h"
+#include "support/bitset.h"
+
+#ifndef wasm_analysis_lattices_powerset2_h
+#define wasm_analysis_lattices_powerset2_h
+
+namespace wasm::analysis {
+
+// A powerset lattice whose elements are sets (represented concretely with type
+// `Set`) ordered by subset.
```

Comment:
Perhaps we could call this a "Set Lattice", that is, a lattice over sets and ordered by the natural subset relation? "Powerset" is a specific mathematical operation to generate all subsets but this lattice here does not create a powerset nor does it require that the elements actually be a powerset IIANM.

If so then the one below could be "Powerset Lattice" (without "finite" in the name, which feels odd there to me).

Reply:
It's true we never materialize the full powerset, but this lattice does represent a powerset. The lattice itself is the powerset, since its elements are subsets of the set of all `T`.  For that reason, and to be consistent with the literature, we should continue calling this a powerset lattice.

Reply:
Fair enough, but why is the latter called "finite" then, if both are finite?

Reply:
The first one is not necessarily finite. There is no bound on the size of the elements it supports.